java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -36,9 +38,27 @@ open class KotlinUsesExtractor(
     val externalClassExtractor: ExternalDeclExtractor,
     val primitiveTypeMapping: PrimitiveTypeMapping,
     val pluginContext: IrPluginContext,
+    val declarationStack: DeclarationStack,
     val globalExtensionState: KotlinExtractorGlobalState
 ) {
 
+    class DeclarationStack {
```

Comment:
I find it somewhat odd that the declaration stack is in the `UsesExtractor`. In my mind it belongs to the file extractor because that's where we extract the declarations.